platform_umbrella/apps/common_core/lib/common_core/et/knative_report.ex

```diff
@@ -3,35 +3,28 @@ defmodule CommonCore.ET.KnativeReport do
 
   use CommonCore, :embedded_schema
 
+  import CommonCore.ET.ReportTools
+
   alias CommonCore.Ecto.Schema
   alias CommonCore.Resources.FieldAccessors
   alias CommonCore.StateSummary
-  alias CommonCore.StateSummary.FromKubeState
 
   batt_embedded_schema do
     field :pod_counts, :map
   end
 
   def new(%StateSummary{knative_services: services} = state_summary) do
+    pods_by_owner = count_pods_by(state_summary, &FieldAccessors.labeled_owner/1)
```

Comment:
Only calculate the owners once (using the same logic) and pass it around.

Reply:
Please add these as comments to the file.